app/controllers/api/students_controller.rb

```diff
@@ -29,7 +29,18 @@ def show
     render json: student
   end
 
-  def lessons
+  def upcoming_lessons
+    if current_student
+      # TODO(shimmy): To make this endpoint accessible to admins, we should
+      # check if the user is simply logged in, not if they're a student.
+      # Also, for more validations, if the user is a student, we should check if params[:id] == current_user.id
+      lessons = Lesson.where(student_id: params[:id])
+      render json: lessons,
+             each_serializer: LessonIndexSerializer,
+             root: "lessons"
+    else
+      redirect_to new_student_session_path
```

Comment:
in the future, this probably shouldn't redirect to new_student_session_path but an authorization error message